### PR TITLE
New suggestion: FrontendMasters DSA course by Primeagen

### PR DESCRIPTION
**Pull request by MartinFiorde**
Free access, great content and commentary quality with hands on exercies

### CODE REVIEW

**Comment by MartinFiorde** [comment]
Hi @mostafatouny how are you doing?

I would lie if i tell you i watch all the other courses, but I’m familiar with MIT ones, and reviewed all of them now to answer you as complete as i can.

This course I’m recommending has half the length from most others, which goes from 18 hours to over 25 hours. 
In this one, during the 9:20 hours it takes, the instructor makes a great wrap up around all the core concepts from DSA, hence the name of the course.

The focus differ from the others courses in several ways:
* order of topics follow a different approach, explaining basic common concepts as normal like initial logic and Big O and other complexity standards, followed by find algorithms and so on, but making changes that help the didactic progression on all topics, making them more approachable for initial students. 
* A big highlight i can bring up is the order as how is explained sort, starting for a less popular type like bubble sort, and not quick sort like most courses, because of it simplicity in code and logic, to construct the base from which it advance to recursion with few " personal logic rules" shared by the instructor that i didn’t saw in other content, to then make the jump to quick sort to integrate all the concepts.
* It somewhat subjective, but the instructor's communication skills make each lesson interesting and get you involved in the topics, this is not something common to find in virtual nor face-to-face academic content.
* contains a lot of hands-on exercises. They are explained in typescript, but personally I made them in java as I watched each lesson, so is not a mandatory language to know to understand the concepts.
* Typescript was chosen as language for the wide utilization from CS beginners, not as much as an optimal language to develop the underline contents (like “real” arrays, queue and stacks implemented with TS native implementation), but even then, this never become an impediment to the understanding.
* All lessons contain a LOT of insights from real working and academic experience from the instructor, which are great for students more advanced in their learning, but avoiding to be blockers for more beginner ones.
* All content support accurate subtitles in english and you can download full transcriptions, with is a great plus too.

TLDR: it's an excellent piece of content to new learners, and also for more experiences ones, which is a balance hard to find.

**Comment by mostafatouny** [comment]
I see it mainly a for-profit course. Would you clarify?

**Comment by mostafatouny** [comment]
At this point, it is highly preferred to support 100% free-access platforms, with no subsidiary monetization. Look for the high quality of [Algorithms Illuminated](https://www.algorithmsilluminated.org/) for example.

**Comment by MartinFiorde** [comment]
Thank you for the feedback! I'll keep this in mind for future contributions. Wishing you continued success with the project

Best regards!

**Comment by mostafatouny** [comment]
We are happy to hear back from you, @MartinFiorde
